plugins/filter_kubernetes/kube_meta.c

```diff
@@ -87,6 +87,63 @@ static int file_to_buffer(const char *path,
     return 0;
 }
 
+/* Run command to get token when token file cannot open */
+static int get_token_with_command(const char *command, 
+                                  char **out_buf, size_t *out_size)
+{
+    FILE *fp;
+    char buf[4096];
+    char *temp;
+    char *res;
+    size_t size = 1;
+    size_t len = 0;
+
+    if (command == NULL) {
+        return -1;
+    }
+    
+    fp = popen(command, "r");
+    if (fp == NULL) {
+        return -1;
+    }
+
+    res = flb_calloc(1, 4096);
+    if (!res) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+    
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
+        len = strlen(buf);
+        temp = realloc(res, size + len);
```

Comment:
Where are we making the temp free?

Reply:
I assume that I make str = temp so we just need to free(str). Correct me if I am wrong..

Reply:
I am not so sure here. Someone with C expertize can help here.

Reply:
This code looks correct to me. `res` and `temp` are the same address. realloc will free and alloc new memory in one step (as needed)